### PR TITLE
DOC: fix multiple warnings in docs build

### PR DESCRIPTION
**Pull request by neutrinoceros**
## PR Summary
This adresses #3369
Currently we're at 37 warnings, 23 of which are similar errors with docstrings in `yt.utilities.lib.cykdtree.kdtree`.
Here I want to focus on fixing everything else.

### CODE REVIEW

**Comment by neutrinoceros** [comment]
We're down to 29 warnings, none of which I know how to solve, so I'm going to leave it at that and open for review.

File: doc/source/examining/loading_data.rst [hunk 1]
```diff
@@ -59,7 +59,7 @@ The functionality requires the package `ratarmount <https://github.com/mxmlnkn/r
 Under the hood, yt will mount the archive as a (read-only) filesystem. Note that this requires the
 entire archive to be read once to compute the location of each file in the archive; subsequent accesses
 will be much faster.
-All archive formats supported by `ratarmount <https://github.com/mxmlnkn/ratarmount>`_ should be loadable, provided
+All archive formats supported by `ratarmount <https://github.com/mxmlnkn/ratarmount>`__ should be loadable, provided
```
Comment (matthewturk) [inline doc/source/examining/loading_data.rst:62]: Why does this need two underscores?

Comment (neutrinoceros) [reply doc/source/examining/loading_data.rst:62]: the warning:
```
WARNING: Duplicate explicit target name: "ratarmount".
```
and the source of this solution (doubling the underscore): https://thomas-cokelaer.info/blog/2020/04/sphinx-raise-a-warning-duplicate-explicit-target-name/

Comment (matthewturk) [reply doc/source/examining/loading_data.rst:62]: Ah, interesting.  OK, thanks.

**Review by matthewturk — APPROVED** [review]

**Pull request merged**